featurebyte/service/base_document.py

```diff
@@ -972,3 +973,35 @@ async def remove_block_modification_by(
             },
             user_id=self.user.id,
         )
+
+    async def update_document_description(
+        self,
+        document_id: ObjectId,
+        description: Optional[str],
+    ) -> None:
+        """
+        Update document at persistent
+
+        Parameters
+        ----------
+        document_id: ObjectId
+            Document ID
+        description: Optional[str]
+            Document description
+        """
+        document = await self.get_document(document_id=document_id)
+
+        # check if document is modifiable
+        self._check_document_modifiable(document=document)
+
+        # update description
+        update_dict = {"description": description}
+        updated_document = self.document_class(**{**document.dict(by_alias=True), **update_dict})
+        if document != updated_document:
+            # only update if change is detected
+            await self.persistent.update_one(
+                collection_name=self.collection_name,
+                query_filter=self._construct_get_query_filter(document_id=document_id),
+                update={"$set": update_dict},
+                user_id=self.user.id,
+            )
```

Comment:
Would be good if we could refactor the common logic between `update_documents` and `update_document_description` so that it is easier to ensure consistency between these two update methods.